SIPS/sip-26.md

```diff
@@ -113,51 +113,78 @@ Keyring API so it can forward signing requests to it through the
 
 #### Account Snaps
 
-In addition to the Keyring API, non-EVM Account Snaps must also implement the
-`resolveAccountAddress` method defined below. It is used by the RPC Router to
-extract the addres of the account that should handle the signing request from
-the request object.
+As part of the Keyring API, non-EVM Account Snaps MUST also implement support
+for the `keyring_resolveAccountAddress` RPC method defined below. It is used
+by the RPC Router to extract the address of the account that should handle
+the signing request from the request object.
 
 ```typescript
-/**
- * Returns the address of the account that should handle the signing request.
- *
- * @param request - The request object.
- * @returns The account address or `undefined` if the address could not be
- * resolved.
- */
-function resolveAccountAddress(request: MultichainRequest): string | undefined;
+type ResolveAccountAddressRequest = {
+  method: "keyring_resolveAccountAddress";
+  params: {
+    chainId: CaipChainId;
+    request: JsonRpcRequest;
+  };
+};
 ```
 
+The implementation MUST return a value of the type `{ address: string }` or `null`.
+
 #### Protocol Snaps
 
 Protocol Snaps implement and expose methods that do not require an account to
 execute and MUST list their supported methods in their manifest file:
 
 ```json5
 "initialPermissions": {
-  "endowment:protocol-methods": {
+  "endowment:protocol": {
     "chains": {
-      "<chain_id_1>": [
+      "<chain_id_1>": {
+        "methods": [
         // List of supported methods
-      ]
+        ]
```

Comment:
should we include a separate property for notifications here?

Reply:
Added!